packages/@aws-cdk/aws-redshift/lib/private/database-query-provider/table.ts

```diff
@@ -48,28 +71,79 @@ async function updateTable(
-    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, clusterProps);
+  if (tableAndClusterProps.clusterName !== oldClusterProps.clusterName || tableAndClusterProps.databaseName !== oldClusterProps.databaseName) {
+    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, tableAndClusterProps);
   }
 
   const oldTableNamePrefix = oldResourceProperties.tableName.prefix;
   if (tableNamePrefix !== oldTableNamePrefix) {
-    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, clusterProps);
+    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, tableAndClusterProps);
   }
 
   const oldTableColumns = oldResourceProperties.tableColumns;
   if (!oldTableColumns.every(oldColumn => tableColumns.some(column => column.name === oldColumn.name && column.dataType === oldColumn.dataType))) {
-    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, clusterProps);
+    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, tableAndClusterProps);
   }
 
-  const additions = tableColumns.filter(column => {
+  const columnAdditions = tableColumns.filter(column => {
     return !oldTableColumns.some(oldColumn => column.name === oldColumn.name && column.dataType === oldColumn.dataType);
   }).map(column => `ADD ${column.name} ${column.dataType}`);
-  await Promise.all(additions.map(addition => executeStatement(`ALTER TABLE ${tableName} ${addition}`, clusterProps)));
+  if (columnAdditions.length > 0) {
+    alterationStatements.push(...columnAdditions.map(addition => `ALTER TABLE ${tableName} ${addition}`));
+  }
+
+  const oldDistStyle = oldResourceProperties.distStyle;
+  if (!oldDistStyle && tableAndClusterProps.distStyle ||
+    oldDistStyle && !tableAndClusterProps.distStyle) {
+    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, tableAndClusterProps);
+  } else if (oldDistStyle !== tableAndClusterProps.distStyle) {
+    alterationStatements.push(`ALTER TABLE ${tableName} ALTER DISTSTYLE ${tableAndClusterProps.distStyle}`);
+  }
+
+  const oldDistKey = getDistKeyColumn(oldTableColumns)?.name;
+  const newDistKey = getDistKeyColumn(tableColumns)?.name;
+  if (!oldDistKey && newDistKey ||
+    oldDistKey && !newDistKey
+  ) {
+    return createTable(tableNamePrefix, tableNameSuffix, tableColumns, tableAndClusterProps);
+  } else if (oldDistKey !== newDistKey) {
+    alterationStatements.push(`ALTER TABLE ${tableName} ALTER DISTKEY ${newDistKey}`);
+  }
+
+  const oldSortKeyColumns = getSortKeyColumns(oldTableColumns);
+  const newSortKeyColumns = getSortKeyColumns(tableColumns);
+  const oldSortStyle = oldResourceProperties.sortStyle;
+  const newSortStyle = tableAndClusterProps.sortStyle;
+  if ((oldSortStyle === newSortStyle && !areColumnsEqual(oldSortKeyColumns, newSortKeyColumns))
+    || (oldSortStyle !== newSortStyle)) {
+    switch (newSortStyle) {
+      case TableSortStyle.INTERLEAVED:
```

Comment:
Can you link to the documentation here describing why one type of switch requires a new table, and others can use an ALTER command?

Reply:
In this doc: https://docs.aws.amazon.com/redshift/latest/dg/r_ALTER_TABLE.html
I couldn't find a explicit ALTER statement for INTERLEAVED sort style.

Reply:
Awesome; can you add that as a comment here, just so its clear in-source?